src/dj.erl

```diff
@@ -395,6 +397,33 @@ uuid(v4) ->
     end,
   chain(binary(), UuidFromBinary).
 
+%% @doc Decode a bounded integer from JSON
+%%
+%% Occasionally, you may want to decode a JSON integer only when it sits between
+%% certain bounds.
+%%
+%% ```
+%%    -spec score() -> dj:decoder(1..10)
+%%    score() ->
+%%        dj:integer(1, 10).
+%%
+%%    {ok, 5} = dj:decode(<<"5">>, score()).
+%%
+%%    E = {custom, {integer_out_of_bounds, 1, 10, 0}},
+%%    {error, [E]} = dj:decode(<<"0", score()).
+%% '''
+%%
+%% @see integer/0
+-spec integer(Min :: integer(), Max :: integer()) -> decoder(integer()).
+integer(Min, Max) when Max < Min ->
+  integer(Max, Min);
+integer(Min, Max) ->
+  CheckBounds =
+    fun (Int) when Min =< Int andalso Int =< Max -> succeed(Int);
```

Comment:
:art: :information_source: I think it would be more readable if we were to write it as:
`when Int >= Min andalso Int =< Max`

Because now we're comparing int to min and max, rather than min to int, and int to max.

Reply:
Yeah, I sort of wanted to visually express that `Int` should be between `Min` and `Max`. In retrospect, it takes less effort to mentally parse your proposed variation 😄 I'll change it, thanks!

Reply:
Ahh I see what you mean by visually expressing that Int is in between. It's just a shame that both operators are `=<`, which kind of throws you off I think